pkg/reconciler/utils/pubsub/subscription.go

```diff
@@ -56,26 +54,20 @@ func (r *Reconciler) ReconcileSubscription(ctx context.Context, id string, subCo
 			return nil, err
 		}
 		if config.Topic != nil && config.Topic.String() == deletedTopic {
-			logger.Error("The topic is deleted")
-			updater.MarkSubscriptionFailed("DeletedTopic", "Pull subscriptions must be recreated to work with recreated topic")
-			return nil, deletedTopicErr
+			logger.Warn("Detected deleted topic. Going to recreate the pull subscription.")
```

Comment:
I think this should still be error level. The error message should warn that data loss may occur if the subscription has unacknowledged messages.